java-checks/src/main/java/org/sonar/java/checks/StringPrimitiveConstructorCheck.java

```diff
@@ -20,51 +20,85 @@
+    MethodMatchers.create().ofTypes("java.lang.Integer").constructor().addParametersMatcher("int").build(),
+    MethodMatchers.create().ofTypes("java.lang.Long").constructor().addParametersMatcher("long").build(),
+    MethodMatchers.create().ofTypes("java.lang.Float").constructor().addParametersMatcher("float").build(),
+    MethodMatchers.create().ofTypes("java.lang.Double").constructor().addParametersMatcher("double").build(),
+    MethodMatchers.create().ofTypes("java.lang.Boolean").constructor().addParametersMatcher("boolean").build());
+
+  private static final Map<String, String> classToLiteral = Map.of(
+    "String", "string",
+    "Double", "double",
+    "Integer", "int",
+    "Boolean", "boolean",
+    "Byte", "byte",
+    "Character", "char",
+    "Short", "short",
+    "Long", "long",
+    "Float", "float"
+    );
+
+
   @Override
-  protected MethodMatchers getMethodInvocationMatchers() {
-    return MethodMatchers.or(
-      MethodMatchers.create()
-        .ofTypes(STRING)
-        .constructor()
-        .addWithoutParametersMatcher()
-        .addParametersMatcher(STRING)
-        .build(),
-      MethodMatchers.create().ofTypes("java.lang.Byte").constructor().addParametersMatcher("byte").build(),
-      MethodMatchers.create().ofTypes("java.lang.Character").constructor().addParametersMatcher("char").build(),
-      MethodMatchers.create().ofTypes("java.lang.Short").constructor().addParametersMatcher("short").build(),
-      MethodMatchers.create().ofTypes("java.lang.Integer").constructor().addParametersMatcher("int").build(),
-      MethodMatchers.create().ofTypes("java.lang.Long").constructor().addParametersMatcher("long").build(),
-      MethodMatchers.create().ofTypes("java.lang.Float").constructor().addParametersMatcher("float").build(),
-      MethodMatchers.create().ofTypes("java.lang.Double").constructor().addParametersMatcher("double").build(),
-      MethodMatchers.create().ofTypes("java.lang.Boolean").constructor().addParametersMatcher("boolean").build(),
-      MethodMatchers.create().ofTypes("java.math.BigInteger").constructor().addParametersMatcher(STRING).build());
+  public List<Kind> nodesToVisit() {
+    return Collections.singletonList(Tree.Kind.NEW_CLASS);
   }
 
   @Override
-  protected void onConstructorFound(NewClassTree newClassTree) {
+  public void visitNode(Tree tree) {
+    NewClassTree newClassTree = (NewClassTree) tree;
     if (newClassTree.classBody() != null) {
       return;
     }
-    if(isBigIntegerPotentiallyBiggerThanLong(newClassTree)) {
+    if (isBigIntegerPotentiallyBiggerThanLong(newClassTree)) {
       return;
     }
-    reportIssue(newClassTree.identifier(), "Remove this \"" + newClassTree.symbolType().name() + "\" constructor");
+    if(matchers.matches(newClassTree)) {
+      QuickFixHelper.newIssue(context)
+        .forRule(this)
+        .onTree(newClassTree.identifier())
+        .withMessage("Remove this \"" + newClassTree.symbolType().name() + "\" constructor")
```

Comment:
`withMessage()` allows you to directly pass parameters. So you can clean it a lot by using the following:
```java
.withMessage("Remove this \"%s\" constructor.", newClassTree.symbolType().name())
```
And you can also extract the issue message constant then.